appformer-form-modeler/appformer-form-modeler-codegen/appformer-form-modeler-codegen-impl/src/main/java/org/kie/appformer/formmodeler/codegen/view/impl/java/RoasterViewSourceGenerator.java

```diff
@@ -37,137 +32,182 @@
-            if (helper == null) continue;
+            final InputCreatorHelper helper = creatorHelpers.get(fieldDefinition.getFieldType().getTypeName());
+            if (helper == null) {
+                continue;
+            }
 
-            if ( extraFieldsEnabled() ) {
-                addExtraFields( helper, context, viewClass, fieldDefinition );
+            if (extraFieldsEnabled()) {
+                addExtraFields(helper,
+                               context,
+                               viewClass,
+                               fieldDefinition);
             }
 
-            final PropertySource<JavaClassSource> property = viewClass.addProperty( getWidgetFromHelper( helper, fieldDefinition), fieldDefinition.getName() );
+            final PropertySource<JavaClassSource> property = viewClass.addProperty(getWidgetFromHelper(helper,
+                                                                                                       fieldDefinition),
+                                                                                   fieldDefinition.getName());
 
             final FieldSource<JavaClassSource> field = property.getField();
             field.setPrivate();
 
-            initializeProperty( helper, context, viewClass,fieldDefinition, field );
-
-            if (helper instanceof RequiresCustomCode ) ((RequiresCustomCode )helper).addCustomCode( fieldDefinition, context, viewClass );
+            initializeProperty(helper,
+                               context,
+                               viewClass,
+                               fieldDefinition,
+                               field);
 
+            if (helper instanceof RequiresCustomCode) {
+                ((RequiresCustomCode) helper).addCustomCode(fieldDefinition,
+                                                            context,
+                                                            viewClass);
+            }
 
-            if ( !StringUtils.isEmpty(fieldDefinition.getBinding()) && !(fieldDefinition instanceof EmbedsForm )) {
+            if (!StringUtils.isEmpty(fieldDefinition.getBinding()) && !(fieldDefinition instanceof EmbedsForm)) {
                 String bindingExpression = context.getFormDefinition().getModel().getName() + "." + fieldDefinition.getBinding();
-                field.addAnnotation( ERRAI_BOUND ).setStringValue( "property", bindingExpression );
+                AnnotationSource<JavaClassSource> boundAnnotation = field.addAnnotation(ERRAI_BOUND);
+                boundAnnotation.setStringValue("property",
+                                               bindingExpression);
+                if (helper instanceof RequiresValueConverter) {
+
+                    String converterClassName = ((RequiresValueConverter) helper).getConverterClassName(fieldDefinition);
+                    if (!StringUtils.isEmpty(converterClassName)) {
```

Comment:
@alessioscalici the logic inside this if should probably also be part of some unit test.